pkg/endpoint/endpoint.go

```diff
@@ -198,6 +199,81 @@ func getCiliumClient() (ciliumClient cilium_client_v2.CiliumV2Interface, err err
 	return ciliumEndpointSyncControllerK8sClient.CiliumV2(), nil
 }
 
+// RunK8sCiliumEndpointSyncGC starts the node-singleton sweeper for
+// CiliumEndpoint objects where the managing node is no longer running. These
+// objects are created by the sync-to-k8s-ciliumendpoint controller on each
+// Endpoint.
+// The general steps are:
+//   - get list of nodes
+//   - only run with probability 1/nodes
+//   - get list of CEPs
+//   - for each CEP
+//       delete CEP if the node does not exist
+// CiliumEndpoint objects follow a nodename-cep-endpointID scheme
+func RunK8sCiliumEndpointSyncGC() {
```

Comment:
k8s 1.9 has added leases to cleanup the apiserver endpoints.
https://github.com/rphillips/kubernetes/commit/08966accfb40b6748b741613f7f7f40e8e24140e

Could we use it as well instead of GC?

Reply:
I'm also fine to stick with GC for the initial version

Reply:
I'm still trying to find how they use the leases. It's really unclear to me if it's a more general concept or specific to this reconciler (unless you were thinking we could pretend each CRD is a "master" of itself). In particular, I'm not clear that anything is deleted vs the annotations acting as a priority ordering when picking a master.

Reply:
I probably don't know more than you at this point. I saw this go in and observed that it addresses the same problem as we have in this case. Resources created by components.

Reply:
@raybejjani can you create a GH issue so we don't lose track of this?

Reply:
yup. https://github.com/cilium/cilium/issues/3101